packages/block/src/block.ts

```diff
@@ -140,6 +140,16 @@ export class Block {
       throw new Error('Invalid serialized block input. Must be array')
     }
 
+    if (
+      opts?.common !== undefined &&
+      opts?.common?.isActivatedEIP(4895) &&
+      (values[3] === undefined || !Array.isArray(values[3]))
+    ) {
+      throw new Error(
```

Comment:
TODO also check if we have this problem in header (and maybe also for `excessDataGas` in header..?)

Reply:
Also check if we should move this into `fromValuesArray` (in header this is the case..?)

Reply:
I do not feel super good that `fromHeaderData` / `fromBlockData` is more "permissive" than the other methods. For instance, if you `fromBlockData` with 4895 (withdrawals) active, then it will not complain and just instantiate the block with an empty withdrawals list. However, currently `fromRLPSerializedBlock` will throw (and probably `fromValuesArray` should also throw). Not super sure what I think about these "two levels" of "security" of these methods.

Reply:
I know what you mean but the thing is that we actually use this *a lot* - in tx there is a similar pattern and we use this for all sorts of convenience instantiations spread over hundreds of places - things like:

```typescript
const tx = FeeMarketEIP1559Transaction.fromTxData(
  { to: B.address, maxFeePerGas: 6 },
  { common }
)
```

This differentiation also *has* some logic, since the other two constructors (RLP, values array) *are* just used for the very most part in contexts (client, devp2p, other) where a "full" tx/block is expected, while - I would assume - the `from*Data` constructor prevails for testing or non-complete setups.

I think we do at least want to keep this more permissive behavior, at least as one option. One option to improve here might be an additional flag, just for this one constructor (or for all constructors but with a different default setting?) where one can set the checks to be more strict or something.

Reply:
I agree, but I think we should indeed add this strict check flag on the `from*Data`, such that users /must/ provide all fields.

Reply:
shouldn't you just call `validateData` here? no reason to duplicate checks